internal/controller/resourcefieldexport/output.go

```diff
@@ -21,22 +23,31 @@ func (r *Reconciler) writeToSecret(ctx context.Context, name string, namespace s
 	for k, v := range values {
 		secretCopy.Data[k] = []byte(v)
 	}
-	return r.Update(ctx, secretCopy)
+	err = r.Update(ctx, secretCopy)
```

Comment:
Instead of logging in the helper functions it would be better if the log were written in the main reconcile function on successful update of the status.